s3.py

```diff
@@ -24,7 +24,17 @@ def list(prefix):
     files = [x['Key'] for x in response['Contents']] if 'Contents' in response else []
     folders = [x['Prefix'] for x in response['CommonPrefixes']] if 'CommonPrefixes' in response else []
 
-    return {'files': files, 'folders': folders}
+    # Apparently you can't get tags from list_objects...
+    tags = {}
+    for f in files:
+        file_tags = client.get_object_tagging(
+            Bucket=BUCKET,
+            Key=f
+        )
+
+        tags[f] = file_tags["TagSet"] or [{'Key': 'public', 'Value': 'False'}]
```

Comment:
Is file_tags a dict?
Is it possible to use `filte_tags.get("TagSet", [{'Key': 'public', 'Value': 'False'}])` instead? I think that is clearer than using or.

Reply:
file_tags is a list of objects

Reply:
How can you index it using a string then?